treescript/src/treescript/script.py

```diff
@@ -30,29 +30,20 @@
     """
     await checkout_repo(config, task, repo_path)
     num_changes = 0
-    for action in actions:
-        if action in ["tagging", "tag"]:
-            num_changes += await do_tagging(config, task, repo_path)
-        elif "version_bump" == action:
-            num_changes += await bump_version(config, task, repo_path)
-        elif "l10n_bump" == action:
-            num_changes += await l10n_bump(config, task, repo_path)
-        elif "push" == action:
-            pass  # handled after log_outgoing
-        else:
-            raise NotImplementedError("Unexpected action")
+    if "tag" in actions:
```

Comment:
I think this is a good set of changes overall.

We had been discussing moving treescript to behaviors rather than actions, which makes sense because certain combinations of actions aren't intended to run in the same task. We should move the logic of which operations to run, and in which order, to the behavior function. I think these changes can land in the meantime, though. Please be mindful of the merge day PR.

Reply:
Yeah, this (partly) grew out of changes I was thinking of suggesting on that PR.